runbotics/runbotics-orchestrator-ui/src/src-app/components/CredentialSelectSet/CredentialSelectSet.tsx

```diff
@@ -0,0 +1,108 @@
+import { ChangeEvent, FunctionComponent, useEffect, useMemo, useState } from 'react';
+
+import { MenuItem, Select, Typography } from '@mui/material';
+import _ from 'lodash';
+import { CredentialDto } from 'runbotics-common';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+
+import { Container, Box } from './CredentialSelectSet.styles';
+
+interface CredentialSelectSetProps {
+    credentials: CredentialDto[];
+    handleCredentialChange: (credential: CredentialDto) => void;
+}
+
+const CredentialSelectSet: FunctionComponent<CredentialSelectSetProps> = ({
+    credentials, handleCredentialChange
+}) => {
+    const { translate } = useTranslations();
+
+    const [author, setAuthor] = useState<string>(null);
+    const credentialsFilteredByAuthor = credentials.filter(
+        (credential) => credential.createdBy.login === author
+    );
+
+    const [collectionName, setCollectionName] = useState<string>(null);
+    const credentialsFilteredByCollection = credentialsFilteredByAuthor.filter(
+        (credential) => credential.collection.name === collectionName
+    );
```

Comment:
maybe useMemo

Reply:
I used it, @socjalis said that is is not necessary so I changed it